suite-native/device/src/deviceMiddleware.ts

```diff
@@ -0,0 +1,70 @@
+import { AnyAction, isAnyOf } from '@reduxjs/toolkit';
+
+import { createMiddlewareWithExtraDeps } from '@suite-common/redux-utils';
+import { DEVICE } from '@trezor/connect';
+import {
+    deviceActions,
+    forgetDisconnectedDevices,
+    handleDeviceConnect,
+    handleDeviceDisconnect,
+    observeSelectedDevice,
+    selectDeviceThunk,
+} from '@suite-common/wallet-core';
+
+const isActionDeviceRelated = (action: AnyAction): boolean => {
+    if (
+        isAnyOf(
+            deviceActions.authDevice,
+            deviceActions.authFailed,
+            deviceActions.selectDevice,
+            deviceActions.receiveAuthConfirm,
+            deviceActions.updatePassphraseMode,
+            deviceActions.addButtonRequest,
+            deviceActions.rememberDevice,
+            deviceActions.forgetDevice,
+        )(action)
+    ) {
+        return true;
+    }
+
+    if (Object.values(DEVICE).includes(action.type)) return true;
+
+    return false;
+};
+
+export const prepareDeviceMiddleware = createMiddlewareWithExtraDeps(
+    (action, { dispatch, next }) => {
+        if (action.type === DEVICE.DISCONNECT) {
+            dispatch(forgetDisconnectedDevices(action.payload));
+        }
+
+        // propagate action to reducers
+        next(action);
```

Comment:
Why is this here? Always put it on the end unless some very specific use cases that must be explicitely desribed in comment then.